src/features/layers_in_area/atoms/areaLayersDetailsResource/areaLayersDetailsParamsAtom.ts

```diff
@@ -30,13 +30,27 @@ export const areaLayersDetailsParamsAtom = createAtom(
 
     const mustBeRequested = allLayers.filter((layer) => {
       const isEnabled = enabledLayers.has(layer.id);
-      if (isEnabled) {
-        const cacheKey = layer.eventIdRequiredForRetrieval ? eventId : null;
-        const cached = cache.get(cacheKey)?.get(layer.id) ?? null;
+      if (!isEnabled) return false;
+
+      if (layer.boundaryRequiredForRetrieval) {
+        const cacheKey: string | undefined = focusedGeometry?.geometry['hash'];
+        const cached = cacheKey ? cache.get(cacheKey) : false;
```

Comment:
what layer you check here?

Reply:
a layer with boundaryRequiredForRetrieval flag

so at first we check if layer is boundaryRequiredForRetrieval
check in cache by hash key

then if no, check if layer eventIdRequiredForRetrieval
if yes - check cache by eventId key
if no - check cache by null key

Reply:
"so at first we check if layer is boundaryRequiredForRetrieval" what layer? You have N such lawyers?

Reply:
it's only 2 of them: 
Kontur Urban Core
Kontur Settled Periphery